packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -873,21 +838,33 @@ export class PromoteService extends BaseService {
             (change) => change.action === PromotionAction.UPDATE,
         );
         const updatedSpacePromises = updatedSpaces.map((spaceChange) =>
-            this.spaceModel.update(spaceChange.data.uuid, spaceChange.data),
+            // Only update name, promotion should not change permissions
+            this.spaceModel.update(spaceChange.data.uuid, {
+                name: spaceChange.data.name,
+            }),
         );
         await Promise.all(updatedSpacePromises);
 
         const newSpaces = spaceChanges
             .filter((change) => change.action === PromotionAction.CREATE)
-            .map((spaceChange) => {
+            .map(async (spaceChange) => {
                 const { data } = spaceChange;
-                return this.spaceModel.createSpace(
+                const space = await this.spaceModel.createSpace(
                     data.projectUuid,
                     data.name,
                     user.userId,
                     data.isPrivate,
                     data.slug,
                 );
+                if (data.isPrivate) {
+                    // user who created the private space is set as a space admin by default
+                    await this.spaceModel.addSpaceAccess(
```

Comment:
why not copying all the spaceAccess for this space ? , so everyone that already has access in dev also get access in prod ?

Reply:
Created a separate ticket to copy all user access [#11469](https://github.com/lightdash/lightdash/issues/11469)